### DISCUSSION TITLE
Support helmreleases uninstall.crd

### DISCUSSION THREAD

**Discussion by xunholy** [fluxcd/flux2#1950] [Proposals] [answered]
This may already be potentially possible but I couldn't see the ability documented in the schema or doc's;

I'd like to be able to remove/uninstall a helmrelease and choose not to also cleanup the CRDs  - removing CRDs consequently will remove all the CRs however these CRs I'd like to persist in cases where it's a multi tenant cluster and other teams own and deploy these CRs or where these CRs have an ownership stanza on an already created resource.

@stefanprodan happy to discuss more specific the edge cases but this would be a nice addition considering the install and upgrade both have the ability to CreateReplace.

**Comment by hiddeco** [comment, marked as answer]
For uninstalls, there is no custom logic at present, and the controller does what Helm does (which is leaving the Custom Resource Definitions untouched):

```console
$ helm repo add bitnami-labs https://bitnami-labs.github.io/sealed-secrets/
"bitnami-labs" already exists with the same configuration, skipping

$ kubectl get crds
No resources found

$ helm install --namespace kube-system my-release bitnami-labs/sealed-secrets
NAME: my-release
LAST DEPLOYED: Fri Oct 15 13:43:52 2021
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
You should now be able to create sealed secrets.
...

$ kubectl get crd
NAME                        CREATED AT
sealedsecrets.bitnami.com   2021-10-15T11:43:51Z

$ helm uninstall --namespace kube-system my-release
release "my-release" uninstalled

$ kubectl get crd
NAME                        CREATED AT
sealedsecrets.bitnami.com   2021-10-15T11:43:51Z
```

**Reply by xunholy** [reply]
Thanks @hiddeco for confirming that, I was under the impression the prune would have cleaned up more than a regular helm uninstall but this is good to know. 

Will you see it always being the case that is won't remove or cleanup CRDs? If so then agreed this wouldn't be needed.

**Reply by hiddeco** [reply]
It will always continue to be the default until Helm decides it isn't.

There is theoretically room to create a feature like the non-default CRD upgrade, but introduction would require a solid scenario and request from the community, and would always be opt-in.
